video/out/vo_dmabuf_wayland.c

```diff
@@ -150,27 +153,78 @@ static bool drmprime_dmabuf_importer(struct mp_image *src, struct wlbuf_pool_ent
 }
 #endif
 
+static void set_viewport_source(struct vo *vo, struct mp_rect src) {
+    struct priv *p = vo->priv;
+    struct vo_wayland_state *wl = vo->wl;
+
+    if (wl->video_viewport) {
+        if ( !mp_rect_equals(&p->src,&src)) {
+
+            // 1. update viewport source
+            wp_viewport_set_source(wl->video_viewport, src.x0 << 8,
+                                    src.y0 << 8, mp_rect_w(src) << 8,
+                                    mp_rect_h(src) << 8);
+            // 2. reset buffer pool
+            p->want_reset = true;
+
+            // 3. update to new src dimensions
+            p->src = src;
+        }
+    }
+}
+
+static void set_video_surface(struct vo_wayland_state *wl, struct mp_rect dst){
+    if (wl->video_viewport)
+        wp_viewport_set_destination(wl->video_viewport, mp_rect_w(dst), mp_rect_h(dst));
+    wl_subsurface_set_position(wl->video_subsurface, dst.x0, dst.y0);
+}
+
+static void set_video_surface_vo(struct vo *vo){
```

Comment:
It feels like `set_video_surface_vo` and `set_video_surface` should just be combined into `resize`. It's conceptually similar and would reduce some of the code duplication.

Reply:
good idea - fixed now.